server/dive_server/crud_dataset.py

```diff
@@ -278,36 +278,37 @@ def get_dataset_query(
     shared: bool,
     level=AccessType.READ,
 ):
-    permissionsClause = Folder().permissionClauses(user=user, level=level)
-    dataset_clause = {f'meta.{constants.DatasetMarker}': {'$in': TRUTHY_META_VALUES}}
-    query_parts = [{'$and': [dataset_clause, permissionsClause]}]
+    base_query = {
+        '$and': [
+            {f'meta.{constants.DatasetMarker}': {'$in': TRUTHY_META_VALUES}},
+            Folder().permissionClauses(user=user, level=level),
+        ]
+    }
+    optional_query_parts: List[Dict[str, Any]] = []
+
     if published:
-        published_query = {f'meta.{constants.PublishedMarker}': {'$in': TRUTHY_META_VALUES}}
-        query_parts.append({'$and': [published_query, dataset_clause]})
+        optional_query_parts.append(
+            {f'meta.{constants.PublishedMarker}': {'$in': TRUTHY_META_VALUES}}
+        )
     if shared:
-        shared_query = [
+        optional_query_parts.append(
             {
-                # Find datasets not owned by the current user
-                '$nor': [
-                    {'creatorId': {'$eq': user['_id']}},
-                    {'creatorId': {'$eq': None}},
-                ],
-            },
-            {
-                # But where the current user still has access
-                'access.users': {
-                    '$elemMatch': {
-                        'id': user['_id'],
-                    }
-                },
-            },
-            dataset_clause,
-        ]
-        query_parts.append({'$and': shared_query})
+                '$and': [
+                    {
+                        # Find datasets not owned by the current user
+                        '$nor': [{'creatorId': {'$eq': user['_id']}}, {'creatorId': {'$eq': None}}]
+                    },
+                    {
+                        # But where the current user still has access
+                        'access.users': {'$elemMatch': {'id': user['_id']}}
+                    },
```

Comment:
Is there a reason not to use `Folder.permissionClauses(user, Access.Read)`  It does a more explicit check on the  level and gives the addition of checking if  the user has groups permissions to read as well?  At least that what it looks like from the source code?

Reply:
It's actually even easier than that.  Because the whole query is now correct, I can remove the permissions clause entirely from this section.

Reply:
okay so this would be disregarding groups then?  Although I don't think we use groups for anything yet.